### DISCUSSION TITLE
Expand specific FluentAccordionItem with button

### DISCUSSION THREAD

**Discussion by bao86** [microsoft/fluentui-blazor#1024] [Q&A]
Hi there,

How I can select specific FluentAccordionItem and change Expanded property ?

```
<FluentAccordion>
    <FluentAccordionItem Expanded="true" Heading="Panel one">
        <FluentButton>Open Next Panel</FluentButton>
    </FluentAccordionItem>
        <FluentAccordionItem Expanded="false" Heading="Panel two">
        Panel two content
    </FluentAccordionItem>
</FluentAccordion>
```

When click on button it will collapse panel one and expand panel two

**Comment by vnbaaij** [comment]
See the example at https://fluentui-blazor.net/Accordion#accordiondefaultexam. The key point is that you need to work with the `ActiveId` (and set the `Id` per item by hand to prevent them from getting created automatically).

**Reply by bao86** [reply]
Thanks but How can I access to component with the id ?

**Reply by vnbaaij** [reply]
You can just assign an `Id` to every item and then set the `ActiveId` on the `FluentAccordion` (based on the button click)

**Reply by bao86** [reply]
I'm really sorry but I don't understand everything :( I'm such a loser

I have something like this : 
![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/cb42722c-cb68-4980-9a20-97b0a95a5d25)

![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/b050ebe7-e724-4b3f-ae24-d0b7de51e528)

**Reply by vnbaaij** [reply]
`ActiveId` needs to be a (string) variable (like  in the example: ActiveId="@activeId")
Then in your `packageNextStep` method you set that variable to the id that you want to go to, ie: activeId = "accor-installation-sequence"

But the code shown on your first image is **not** going to work. You have an accordion item *inside* another accordion item. That is not how it is supposed to be used. If you would want to nest them (which I would not advise to do so), then there should be another FluentAccordion in the `ChildContent` first. But you are also having only 1 item (accor-package-info) in that child content so why is that  an Accordion item?

p.s You're not a loser! The key to understanding is asking questions. A loser would not ask the questuions and just give up,..

**Reply by bao86** [reply]
accordion item inside another seems working : 
![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/90784ccd-349e-4fae-94cd-43c58a62f43d)


I miss something, when i click on Next button (call packageNextStep method), nothing happen :( 

![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/fe29daa6-88f4-49a7-abe8-fb80db78db5d)

![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/ef99b036-94de-4ed2-9496-8a28c40eecf2)

**Reply by vnbaaij** [reply]
No, it is not working... It just displays as you expect. Might very well be what is causing the issue with the button not working

**Reply by bao86** [reply]
oh ok. I will change it

**Reply by bao86** [reply]
So now I don't have accordion item in another accordion item but still not working :( Don't understand

![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/98739cf2-8a70-4fc9-bc27-9252f4e504a8)
![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/9b134af7-c2e1-4423-a450-6f36c2fe2b63)
![image](https://github.com/microsoft/fluentui-blazor/assets/103753661/cc942cf0-d431-496c-883f-dc5559bab865)

![2023-11-27_23-44-50](https://github.com/microsoft/fluentui-blazor/assets/103753661/7cfd5d36-4328-4546-b340-503571effc17)

**Reply by vnbaaij** [reply]
Creeate a repository, post the code there and then we can take a look. It is not helpful to dubug from images.

**Reply by bao86** [reply]
You can download the project here
https://github.com/bao86/Debug

**Reply by vnbaaij** [reply]
Line 6: <FluentAccordion @bind-ActiveId="@activeId">
Line 10:  <FluentAccordionItem Id="accor-package-info" Expanded="info_expanded">
Line 62:  <FluentAccordionItem Id="accor-inst-welcome-prompt" Expanded="welcome_expanded">

Beginning of code block add:
    bool info_expanded = true;
    bool welcome_expanded = false;

At the end:
private void packageNextStep()
{
    activeId = "accor-inst-welcome-prompt";
    welcome_expanded = true;
    info_expanded = false;

}

Just a guideline, adapt to your own needs...

**Reply by bao86** [reply]
Thank you :)
If I understand 

Line 6 : `<FluentAccordion @bind-ActiveId="@activeId">`
Line 10 : `<FluentAccordionItem Id="accor-package-info" Expanded="@info_expanded">`
Line 62 : `<FluentAccordionItem Id="accor-inst-welcome-prompt" Expanded="@welcome_expanded">`

beginning of code block :
```
string activeId;
bool info_expanded = true;
bool welcome_expanded = false;
```

in the method : 
```
private void packageNextStep()
{
    activeId = "accor-inst-welcome-prompt";
    welcome_expanded = true;
    info_expanded = false;
}
```

It's working fine :) thank you again but it works once.
If I collapase the "Welcome" Panel and then Expand "Info" Panel and finally click on "Next" again, is not working :( 
Have to reload the page

![2023-11-28_12-31-30](https://github.com/microsoft/fluentui-blazor/assets/103753661/996bd4a0-eee4-4948-9191-e73fb5f18c11)

**Reply by vnbaaij** [reply]
You need to reset the booleans when you cahange back. That is more general coding and not specifi to the library. You need to develop your own skills there

**Reply by bao86** [reply]
Thanks for your time and help :)
